### PR TITLE
Re-enable MHC tests 

### PR DESCRIPTION
**Pull request by cynepco3hahue**
- update machine-api-operator dependency
- create tech preview feature gate
- remove skip directive

### CODE REVIEW

File: pkg/e2e/machinehealthcheck/machinehealthcheck.go [hunk 1]
```diff
@@ -57,9 +58,26 @@ var _ = Describe("[Feature:MachineHealthCheck] MachineHealthCheck controller", f
...
+		// TODO: remove once we can create or update kubemark machines
+		// that will give use possibility to make this test work
+		if isKubemarkProvider {
+			glog.V(2).Info("Can not run this tests with the 'KubeMark' provider")
+			Skip("Can not run this tests with the 'KubeMark' provider")
+		}
+
+		err = e2e.CreateTechPreviewFeatureGate()
+		Expect(err).ToNot(HaveOccurred())
+
```
Comment (enxebre) [inline pkg/e2e/machinehealthcheck/machinehealthcheck.go]: why are we deleting the deployment here?

**Comment by enxebre** [comment]
how long does it take to run the test? before merging this back we should to evaluate https://jira.coreos.com/browse/CLOUD-434

File: pkg/e2e/machinehealthcheck/machinehealthcheck.go (hunk 1 above)
Comment (cynepco3hahue) [reply pkg/e2e/machinehealthcheck/machinehealthcheck.go]: to make machine-api-operator reconcile happen faster and to add machine-health-check container

Comment (enxebre) [reply pkg/e2e/machinehealthcheck/machinehealthcheck.go]: shouldn't it be enough to `CreateTechPreviewFeatureGate`?

Comment (cynepco3hahue) [reply pkg/e2e/machinehealthcheck/machinehealthcheck.go]: I think it should, but will make harder to monitor when the updated deployment is ready and can introduce some flakiness, so I prefer to explicitly delete the deployment

Comment (enxebre) [reply pkg/e2e/machinehealthcheck/machinehealthcheck.go]: why you need to check when the deployment is ready? when the new controller comes up should see the created hc resource, right? Why flakiness? The tests should reproduce the expected user interaction, if some of this does not work there's a bug

Comment (cynepco3hahue) [reply pkg/e2e/machinehealthcheck/machinehealthcheck.go]: done

**Comment by cynepco3hahue** [comment]
@enxebre Can you please take a look?

**Comment by cynepco3hahue** [comment]
For the reference https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-api-actuator-pkg/70/pull-ci-openshift-cluster-api-actuator-pkg-master-e2e-aws-operator/318 the lane that ran with MHC tests.

**Comment by cynepco3hahue** [comment]
@ingvagabund The PR is ready for review.
@enxebre Now that we have a separate target to run `TechPreview` tests, can you please remove the `hold` label.

File: pkg/e2e/framework/machinehealthcheck.go [hunk 2]
```diff
@@ -118,3 +121,29 @@ func StopKubelet(nodeName string) error {
...
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			fg.Name = operator.MachineAPIFeatureGateName
+			fg.Namespace = TestContext.MachineApiNamespace
+			fg.Spec = osconfigv1.FeatureGateSpec{FeatureSet: osconfigv1.TechPreviewNoUpgrade}
+			return client.Create(context.TODO(), fg)
+		}
+		return err
+	}
+	fg.Spec = osconfigv1.FeatureGateSpec{FeatureSet: osconfigv1.TechPreviewNoUpgrade}
```
Comment (ingvagabund) [inline pkg/e2e/framework/machinehealthcheck.go]: You might also check if `osconfigv1.FeatureGateSpec.FeatureSet` is already set to `osconfigv1.TechPreviewNoUpgrade` and `return nil`.

File: pkg/e2e/framework/deloyment.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		return true, nil
+	}); err != nil {
+		glog.Errorf("Error checking isDeploymentAvailable: %v", err)
+		return false
+	}
+	return true
+}
+
+// IsDeploymentHasContainer returns true if the deployment has container with the specified name
+func IsDeploymentHasContainer(deployment *kappsapi.Deployment, containerName string) bool {
```
Comment (ingvagabund) [inline pkg/e2e/framework/deloyment.go]: `s/IsDeploymentHasContainer/DeploymentHasContainer`

File: pkg/e2e/framework/machinehealthcheck.go [hunk 4]
```diff
@@ -118,3 +121,29 @@ func StopKubelet(nodeName string) error {
...
+		Name:      operator.MachineAPIFeatureGateName,
+		Namespace: TestContext.MachineApiNamespace,
+	}
+	err = client.Get(context.TODO(), key, fg)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			fg.Name = operator.MachineAPIFeatureGateName
+			fg.Namespace = TestContext.MachineApiNamespace
+			fg.Spec = osconfigv1.FeatureGateSpec{FeatureSet: osconfigv1.TechPreviewNoUpgrade}
+			return client.Create(context.TODO(), fg)
```
[the remark this answers is not in the document]
Comment (cynepco3hahue) [reply pkg/e2e/framework/machinehealthcheck.go]: done

File: pkg/e2e/framework/machinehealthcheck.go (hunk 2 above)
Comment (cynepco3hahue) [reply pkg/e2e/framework/machinehealthcheck.go]: done

File: pkg/e2e/framework/deloyment.go (hunk 3 above)
Comment (cynepco3hahue) [reply pkg/e2e/framework/deloyment.go]: done

**Pull request merged**